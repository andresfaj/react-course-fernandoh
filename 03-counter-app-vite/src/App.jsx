import PropTypes from "prop-types";

export const App = () => {
  return <div>App</div>;
};

export const FirstApp = ({ title, subTitle, name }) => {
  return (
    <>
      <h1 data-testid="test-title"> {title} </h1>
      <p>{subTitle}</p>
      <p>{subTitle}</p>
      <p>{name}</p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

FirstApp.defaultProps = {
  name: "Andres Achury",
  // title: "NO TITLE FOUND",
  subTitle: "No hay subtitulo",
};
