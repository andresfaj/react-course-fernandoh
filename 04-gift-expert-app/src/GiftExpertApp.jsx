import { useState } from "react";

const GiftExpertApp = () => {
  const [categories, setCategories] = useState(["One Punch"]);

  const handleAddCategory = () => {
    setCategories([...categories, "GTA"]);
  };

  return (
    <>
      <h1>GiftExpertApp</h1>

      <button onClick={handleAddCategory}>Add</button>
      <ol>
        {categories.map((categorie, index) => (
          <li key={`categorie-${index}`}>{categorie}</li>
        ))}
      </ol>
    </>
  );
};

export default GiftExpertApp;
