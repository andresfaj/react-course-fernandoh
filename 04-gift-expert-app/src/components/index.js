export * from "./AddCategory";
export * from "./GifGrid";
export * from "./GiftItem";
