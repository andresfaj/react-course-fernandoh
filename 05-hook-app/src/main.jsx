import React from "react";
import ReactDOM from "react-dom/client";
import { CounterApp } from "./01-useState/CounterApp";
import { CounterWithCustomHook } from "./01-useState/CounterWithCustomHook";
import { SimpleForm } from "./02-useEffect/SimpleForm";
import { HooksApp } from "./HooksApp";
import "./index.css";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <HooksApp /> */}
    {/* <CounterApp /> */}
    {/* <CounterWithCustomHook /> */}
    <SimpleForm />
  </React.StrictMode>
);
