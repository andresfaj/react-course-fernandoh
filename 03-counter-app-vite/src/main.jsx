import React from "react";
import ReactDOM from "react-dom/client";
import "./styles.css";

import { CounterApp } from "./CounterApp";
import { FirstApp } from "./App";

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/* <CounterApp value={12} /> */}
    <FirstApp title="Hola, Soy Vegeta" />
  </React.StrictMode>
);
