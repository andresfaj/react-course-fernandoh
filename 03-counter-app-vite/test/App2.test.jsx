import { render, screen } from "@testing-library/react";
import { FirstApp } from "../src/App";

describe("Test on FirstApp", () => {
  const title = "Hola, soy goku";
  const subTitle = "Soy un subtitulo";

  test("Must match with the snapshot", () => {
    const { container } = render(<FirstApp title={title} />);
    expect(container).toMatchSnapshot();
  });

  test("Must show the message 'Hola soy goku'", () => {
    render(<FirstApp title={title} />);

    expect(screen.getByText(title)).toBeTruthy();

    //screen.debug() to see the object, el screen esta actualizado con los cambios que sucedan en el elemento
  });

  test("Must show the title into a H1", () => {
    render(<FirstApp title={title} />);
    expect(screen.getByRole("heading", { level: 1 }).innerHTML).toContain(
      title
    );
  });

  test("Must show the subtitle sended by props", () => {
    render(<FirstApp title={title} subTitle={subTitle} />);
    expect(screen.getAllByText(subTitle).length).toBe(2);
  });
});
