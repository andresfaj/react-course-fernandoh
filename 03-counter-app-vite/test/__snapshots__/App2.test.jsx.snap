// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on FirstApp Must match with the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola, soy goku
     
  </h1>
  <p>
    No hay subtitulo
  </p>
  <p>
    No hay subtitulo
  </p>
  <p>
    Andres Achury
  </p>
</div>
`;
